### PR TITLE
Ca rotation

### LINKED ISSUE #488: Document how custom CA rotation should be done
As KaaS operator, I would like to have documentation that explains how the custom CA rotation should be done, e.g. when the certificate expires or changes.

### PR DESCRIPTION
**Pull request by chess-knight**
PR #460 added support for OpenStack custom CA certificate.
This PR complement it with easily manageable cacert which is already defined in secret *${CLUSTER_NAME}-cloud-config* so we have only one source of truth. Also, documentation about rotation is created.

Fixes #488

### CODE REVIEW

**Review by fdobrovolny — APPROVED** [review]
LGTM,

However :smile: Is it possible to have two CAs? I can imagine a situation when I want to rotate the certificate, and it can take some time to do everywhere, and I don't want it to affect the cluster. So before the certificate expires, I add the old and new certificates and then I can remove the old one.

So does something like a "zero" downtime certification rotation exist?

File: doc/usage/custom-ca.md [hunk 1]
```diff
@@ -27,19 +27,47 @@ Here the file `ca-bundle.pem` contains custom root CA and potentially intermedia
...
+   ```
 4. When the creation of the workload cluster starts, *~/cluster-defaults/cluster-template.yaml*
    is copied into workload cluster directory (*~/$CLUSTER_NAME/*)
 5. Then the cacert file content is base64 encoded and saved in OPENSTACK_CLOUD_CACERT_B64 variable
    inside *~/$CLUSTER_NAME/clusterctl.yaml*, so it can be used during
    the workload cluster templating
 6. Later, when the workload cluster templates are applied to the management cluster,
-   secret with base64 encoded cacert is created and used by CAPO
+   secret *${CLUSTER_NAME}-cloud-config* with base64 encoded cacert is created and used by CAPO
 7. CAPO will create workload cluster (thanks to steps 5. and 6.) and cacert is
```
[the remark this answers is not in the document]
Comment (chess-knight) [reply doc/usage/custom-ca.md:53]: Thanks, see 7d8e75d61d0b842a38532433b915fd74cfca70cd, I mentioned it in step n. 4

**Comment by chess-knight** [comment]
> Is it possible to have two CAs? I can imagine a situation when I want to rotate the certificate, and it can take some time to do everywhere, and I don't want it to affect the cluster. So before the certificate expires, I add the old and new certificates and then I can remove the old one.
> 
> So does something like a "zero" downtime certification rotation exist?

I added it in 207a4ca559e88c8bb02674104ffbb39659bdd39a. You can have multiple certificates in one file.

**Review by fdobrovolny — APPROVED** [review]
Thank you.

**Pull request merged**